pokemon_search/src/App.js

```diff
@@ -35,7 +29,13 @@ export default function App() {
       .finally(() => {
         setLoading(false);
       });
-  }, []);
+  }, [offset]);
+
+  const fetchAll = async (urls) => {
+    const res = await Promise.all(urls.map((u) => fetch(u)));
+    const jsons = await Promise.all(res.map((r) => r.json()));
```

Comment:
This will wait twice: the first time for all fetches to come back and the second time for all data to be parsed into JSON. 

Alternatively, we could wait for all requests once by combining fetch+parse data steps:

```ts
const jsonsPromises = urls.map(async (u) => {
  const res = await fetch(u);
  const json = await res.json();
  return json;
}
const jsons = await Promise.all(jsonsPromises);
setData(jsons);
```